verify.go

```diff
@@ -154,3 +173,61 @@ func Verify(ctx context.Context, policyEnvelope dsse.Envelope, policyVerifiers [
 
 	return accepted, nil
 }
+
+func verifyPolicySignature(ctx context.Context, vo verifyOptions) error {
```

Comment:
I put this here, but it feels as though it is the wrong place for it.

Do you think this is an overthought or is there a good place to put this function? @mikhailswift @colek42.

Reply:
With the use of `verifyOptions`, this seems like a reasonable place.